satellite/peer.go

```diff
@@ -346,6 +420,11 @@ func (peer *Peer) Close() error {
 		errlist.Add(peer.Kademlia.RoutingTable.SelfClose())
 	}
 
+	// if peer.Kademlia.ndb != nil || peer.Kademlia.kdb != nil {
```

Comment:
Yes, we don't actually want to leak routing table. However, closing the underlying database causes a race with kademlia, since closing kademlia doesn't wait for all lookups to be returned... there probably are other subsystems that need to wait in Close to finish all pending jobs.

i.e. not pretty, but needs to be fixed.